Symfony/CS/Tests/AbstractIntegrationTest.php

```diff
@@ -0,0 +1,260 @@
+        $this->assertTrue(is_dir($fixturesDir), sprintf('Given fixture dir "%s" is not a directory.', $fixturesDir));
+
+        $tests = array();
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($fixturesDir), \RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!preg_match('/\.test$/', $file)) {
+                continue;
+            }
+
+            $test = file_get_contents($file->getRealpath());
+            $fileName = $file->getFileName();
+            if (preg_match('/--TEST--[\r\n](.*?)\s--CONFIG--[\r\n](.*?)\s--INPUT--[\r\n](.*?[\r\n]*)(?:[\r\n]--EXPECT--\s(.*)|$)/s', $test, $match)) {
+                $testTitle = $match[1];
+                $config = $this->getFixersFromConfig($fileName, $match[2]);
+                $input = $match[3];
+                $expected = isset($match[4]) ? $match[4] : null;
+                $tests[] = array($fileName, $testTitle, $config, $input, $expected);
+            } else {
+                throw new InvalidArgumentException(sprintf('Test "%s" is not valid.', $fileName));
+            }
+        }
+
+        $this->assertNotEmpty($tests, sprintf('No tests found in fixtures dir "%s".', $fixturesDir));
+
+        return $tests;
+    }
+
+    /**
+     * Returns the full path to directory which contains the tests.
+     *
+     * @return string
+     */
+    abstract protected function getFixturesDir();
+
+    /**
+     * Returns the full path to directory where the tests will writes the temporary file.
+     *
+     * @return string
+     */
+    abstract protected function getTempDir();
+
+    /**
+     * @param string           $testFileName
+     * @param string           $testTitle
+     * @param FixerInterface[] $fixers
+     * @param string           $input
+     * @param string|null      $expected
+     */
+    protected function doTestIntegration($testFileName, $testTitle, $fixers, $input, $expected = null)
+    {
+        $errorsManager = new ErrorsManager();
+        $fixer = new Fixer();
+        $fixer->setErrorsManager($errorsManager);
+        $fs = new Filesystem();
+
+        $dir = $this->getTempDir();
+        $tmpDir = new \SplFileInfo($dir);
+        if (!$tmpDir->isDir()) {
+            $fs->mkdir($tmpDir);
+        }
+
+        $this->assertTrue($tmpDir->isDir(), sprintf('Given temp dir "%s" is not a directory.', $dir));
+        $this->assertTrue($tmpDir->isWritable(), sprintf('Given temp dir "%s" is not writable.', $dir));
+
+        $tmpFile = $tmpDir->getRealPath().'/tmp.php';
+        $this->assertNotFalse(@file_put_contents($tmpFile, $input), sprintf('Failed to write to tmp. file "%s".', $tmpFile));
+
+        $changed = $fixer->fixFile(new \SplFileInfo($tmpFile), $fixers, false, true, new FileCacheManager(false, null, $fixers));
+
+        if (null !== $expected) {
+            // read back the changed file content
+            $newContent = file_get_contents($tmpFile);
```

Comment:
if I do that it would be after `$fs->remove($tmpFile)`,
I guess this can be done after I refactored towards your RFC here  https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1384#discussion_r38820200